apps/balancer-tools/src/app/apr/(utils)/rewardsApr.ts

```diff
@@ -0,0 +1,124 @@
+import { PoolRewardsQuery } from "@bleu-balancer-tools/gql/src/balancer-rewards/__generated__/Ethereum";
+import { NetworkChainId } from "@bleu-balancer-tools/utils";
+
+import { rewards } from "#/lib/gql/server";
+import { ArrElement, GetDeepProp } from "#/utils/getTypes";
+
+import { isTimePeriodsIntersecting } from "../api/(utils)/date";
+import { fetchPoolAveragesForDateRange } from "./calculatePoolStats";
+
+type RewardInfo = ArrElement<
+  GetDeepProp<PoolRewardsQuery, "rewardTokenDeposits">
+>;
+
+export async function getRewardsAprForDateRange(
+  //TODO: remove unused params on BAL-972
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  poolId: string,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  network: string,
+  startAt: number,
+  endAt: number,
+) {
+  //TODO: remove temp variables on BAL-972
+  const tempPoolId =
+    "0x17e7d59bb209a3215ccc25fffef7161498b7c10d000200000000000000000020";
+  const tempNetwork = String(NetworkChainId.BASE);
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [tvl, volume, symbol, tokenBalance, bptPrice] =
+    await fetchPoolAveragesForDateRange(
+      tempPoolId,
+      tempNetwork,
+      startAt,
+      endAt,
+    );
+
+  const tokensRewards = await getTokensRewards({
+    startAt,
+    endAt,
+    poolId: tempPoolId,
+    network: tempNetwork,
+  });
+
+  if (tokensRewards.length === 0) {
+    return [];
+  }
+
+  const rewards = tokensRewards.map((tokenReward) => {
+    const tokenAmount = tokenReward.amount;
+    // reference for 10_000 https://github.com/balancer/balancer-sdk/blob/f4879f06289c6f5f9766ead1835f4f4b096ed7dd/balancer-js/src/modules/pools/apr/apr.ts#L367
+    const tokenApr = ((tokenAmount * bptPrice) / tvl) * 10_000 * 7;
+    return {
+      token: tokenReward.token,
+      amount: tokenAmount,
+      bptPrice,
+      liquidity: tvl,
+      apr: tokenApr,
+    };
+  });
+  return rewards;
+}
+
+export async function getTokensRewards({
+  startAt,
+  endAt,
+  poolId,
+  network,
+}: {
+  startAt: number;
+  endAt: number;
+  poolId: string;
+  network: string;
+}) {
+  const response = await rewards.gql(network).PoolRewards({
+    poolId,
+  });
```

Comment:
can't we filter by time here as well?

Reply:
we can filter to have a smaller time range to work with, which will be helpful in the future. Right now we have very current rewards. But if you are referring to the actual rewards, then no. I played around with the start and finish periods and couldn't get a query that filters correctly for all cases.

We have 3 cases:
- our date range is inside the reward range
- our date range gets the beginning of a reward range
- our date range gets the end of a reward range

Reply:
we can report that to @mendesfabio if the filtering not working correctly, but perhaps we could also do a time travel query to the end of our interval to avoid fetching all the rewards from a pool and doing all this filtering from memory?

Reply:
I think it is working correctly, but in each case the date conditions are different. I tested this scenario (with multiple days just to make it less confusing):

Reward range: 31/08/2023 - 07/09/2023 (start - finish)

1 Case - 29/08/2023 - 01/09/2023
29/08/2023 lte start -> didn't catch
29/08/2023 gte start -> catch

01/09/2023 lte finish -> didn't catch
01/09/2023 gte finish -> catch

2 Case - 01/09/2023 - 05/09/2023
01/09/2023  lte start -> catch
01/09/2023  gte start -> didn't catch

05/09/2023 lte finish -> didn't catch
05/09/2023 gte finish -> catch

3 Case - 01/09/2023 - 08/09/2023
01/09/2023  lte start -> catch
01/09/2023  gte start -> didn't catch

08/09/2023 lte finish ->catch
08/09/2023 gte finish ->  didn't catch

4 Case - 31/08/2023 - 01/09/2023
31/08/2023  lte start -> didn't catch
31/08/2023  gte start ->  catch

01/09/2023 lte finish -> didn't catch
01/09/2023 gte finish ->  catch

5 Case - 07/09/2023 - 08/09/2023
07/09/2023  lte start -> catch
07/09/2023  gte start ->  didn't  catch

08/09/2023 lte finish -> catch
08/09/2023 gte finish ->  didn't catch

Reply:
idk if I missing something, if so glad to hear a solution kkkkkkk

Reply:
to get all of this cases at once I used `doIntervalsIntersect()`